api/v1beta1/types_class.go

```diff
@@ -458,6 +458,10 @@ type NetworkClassSpec struct {
 	// PrivateDNSZoneName defines the zone name for the Azure Private DNS.
 	// +optional
 	PrivateDNSZoneName string `json:"privateDNSZoneName,omitempty"`
+
+	// PrivateDNSZoneResourceGroup defines the resource group to be used for Azure Private DNS Zone.
+	// +optional
+	PrivateDNSZoneResourceGroup string `json:"privateDNSZoneResourceGroup,omitempty"`
```

Comment:
We should make the description of the field clearer so that the user is not left in limbo, guessing if user should fill out `privateDNSZoneResourceGroup` or not. 
```suggestion
	// PrivateDNSZoneResourceGroup specifies the resource group to be used for Azure Private DNS Zone. 
	// If not specified, the resource group of the cluster will be used to create the Azure Private DNS Zone. 
	// +optional
	PrivateDNSZoneResourceGroup string `json:"privateDNSZoneResourceGroup,omitempty"`
```

Reply:
Got it. I have update the description for this field.